viadot/utils.py

```diff
@@ -117,3 +126,55 @@ def get_flow_last_run_date(flow_name: str) -> str:
         pendulum.parse(last_run_date_raw_format).format("YYYY-MM-DDTHH:MM:SS") + "Z"
     )
     return last_run_date
+
+
+@curry
+def custom_state_handler(
+    tracked_obj: Union["Flow", "Task"],
+    old_state: "prefect.engine.state.State",
+    new_state: "prefect.engine.state.State",
+    only_states: list = None,
+    API_KEY: str = None,
+) -> "prefect.engine.state.State":
+    """
+    Custom state handler configured to work with sendgrid.
+    Works as a standalone state handler, or can be called from within a custom state handler.
+    Args:
+        - tracked_obj (Task or Flow): Task or Flow object the handler is registered with
+        - old_state (State): previous state of tracked object
+        - new_state (State): new state of tracked object
+        - only_states ([State], optional): similar to `ignore_states`, but instead _only_
+            notifies you if the Task / Flow is in a state from the provided list of `State`
+            classes
+    Returns:
+        - State: the `new_state` object that was provided
+    Raises:
+        - ValueError: if the email notification fails for any reason
+
+    """
```

Comment:
This docstring doesn't match to `autoDocstring - Python Docstring Generator` style. If you don't use this extension in VSCode, install it.